From df820df672753b3dc2f4e443c75b88b12bee1062 Mon Sep 17 00:00:00 2001
From: Jason Mobarak <jason@swiftnav.com>
Date: Thu, 8 Feb 2018 22:24:39 -0800
Subject: [PATCH] Fix GCC7 compiler issue

---
 src/gdb_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gdb_main.c b/src/gdb_main.c
index 59d763b..995b62c 100644
--- a/src/gdb_main.c
+++ b/src/gdb_main.c
@@ -103,6 +103,9 @@ void sigusr1_handler(int x)
 
 int gdb_main_loop(struct target_controller *tc, bool in_syscall)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+
 	int size;
 	bool single_step = false;
 
@@ -314,6 +317,7 @@ int gdb_main_loop(struct target_controller *tc, bool in_syscall)
 			gdb_putpacketz("");
 		}
 	}
+#pragma GCC diagnostic pop
 }
 
 static void
-- 
2.15.1

