commit 822ad58a48575fcca44ad5d3f2c2b6996bac444a
Author: Matt Woodward <matthew.woodward@swift-nav.com>
Date:   Thu Jan 24 12:11:21 2019 -0800

    Infinite loop when initialising rpmsg buffer

diff --git a/drivers/rpmsg/virtio_rpmsg_bus.c b/drivers/rpmsg/virtio_rpmsg_bus.c
index a7394e65a49c..f6db67a103d4 100644
--- a/drivers/rpmsg/virtio_rpmsg_bus.c
+++ b/drivers/rpmsg/virtio_rpmsg_bus.c
@@ -210,7 +210,7 @@ rpmsg_sg_init(struct scatterlist *sg, void *cpu_addr, unsigned int len)
 			    offset_in_page(cpu_addr));
 	} else {
 		WARN_ON(!virt_addr_valid(cpu_addr));
-		rpmsg_sg_init(sg, cpu_addr, len);
+		sg_init_one(sg, cpu_addr, len);
 	}
 }
