commit 851119743e73a891addb64f8505b2f3bf37e7aa0
Author: Matt Woodward <matthew.woodward@swift-nav.com>
Date:   Thu Jan 24 12:30:35 2019 -0800

    Handle ELF carveout region before vring rscs

diff --git a/drivers/remoteproc/remoteproc_core.c b/drivers/remoteproc/remoteproc_core.c
index fdfc66d4234a..42ec7cb399b1 100644
--- a/drivers/remoteproc/remoteproc_core.c
+++ b/drivers/remoteproc/remoteproc_core.c
@@ -768,7 +768,6 @@ static int rproc_handle_rproc_mem(struct rproc *rproc,
  * enum fw_resource_type.
  */
 static rproc_handle_resource_t rproc_loading_handlers[RSC_LAST] = {
-	[RSC_CARVEOUT] = (rproc_handle_resource_t)rproc_handle_carveout,
 	[RSC_DEVMEM] = (rproc_handle_resource_t)rproc_handle_devmem,
 	[RSC_TRACE] = (rproc_handle_resource_t)rproc_handle_trace,
 	[RSC_VDEV] = (rproc_handle_resource_t)rproc_count_vrings,
@@ -782,6 +781,10 @@ static rproc_handle_resource_t rproc_rproc_mem_handler[RSC_LAST] = {
 	[RSC_RPROC_MEM] = (rproc_handle_resource_t)rproc_handle_rproc_mem,
 };
 
+static rproc_handle_resource_t rproc_elf_carveout_handler[RSC_LAST] = {
+	[RSC_CARVEOUT] = (rproc_handle_resource_t)rproc_handle_carveout,
+};
+
 /* handle firmware resource entries before booting the remote processor */
 static int rproc_handle_resources(struct rproc *rproc, int len,
 				  rproc_handle_resource_t handlers[RSC_LAST])
@@ -1044,6 +1047,13 @@ static int rproc_fw_boot(struct rproc *rproc, const struct firmware *fw)
 		goto clean_up_resources;
 	}
 
+  /* look for ELF carveout regions, allocate before vrings */
+  ret = rproc_handle_resources(rproc, tablesz, rproc_elf_carveout_handler);
+  if (ret) {
+    dev_err(dev, "Failed to handle elf carveout resources: %d\n", ret);
+    goto clean_up_resources;
+  }
+
 	/* look for virtio devices and register them */
 	ret = rproc_handle_resources(rproc, tablesz, rproc_vdev_handler);
 	if (ret) {
