commit 1ef6e4df5dc6fd62075eb8bf6a06dd4df866fc15
Author: Matt Woodward <matthew.woodward@swift-nav.com>
Date:   Thu Jan 31 17:11:07 2019 +1100

    Bring down CPU1 earlier when loading remoteproc

diff --git a/drivers/remoteproc/zynq_remoteproc.c b/drivers/remoteproc/zynq_remoteproc.c
index d4eac41f9bfa..27b6e2b62849 100644
--- a/drivers/remoteproc/zynq_remoteproc.c
+++ b/drivers/remoteproc/zynq_remoteproc.c
@@ -115,12 +115,6 @@ static int zynq_rproc_start(struct rproc *rproc)
 	dev_dbg(dev, "%s\n", __func__);
 	INIT_WORK(&workqueue, handle_event);
 
-	ret = cpu_down(1);
-	/* EBUSY means CPU is already released */
-	if (ret && (ret != -EBUSY)) {
-		dev_err(dev, "Can't release cpu1\n");
-		return ret;
-	}
 
 	ret = zynq_cpun_start(rproc->bootaddr, 1);
 	/* Trigger pending kicks */
@@ -295,6 +289,13 @@ static int zynq_remoteproc_probe(struct platform_device *pdev)
 	char mem_name[16];
 	int i;
 
+	ret = cpu_down(1);
+	/* EBUSY means CPU is already released */
+	if (ret && (ret != -EBUSY)) {
+		dev_err(&pdev->dev, "Can't release cpu1\n");
+		return ret;
+	}
+
 	rproc = rproc_alloc(&pdev->dev, dev_name(&pdev->dev),
 		&zynq_rproc_ops, NULL,
 		sizeof(struct zynq_rproc_pdata));
@@ -422,6 +423,8 @@ static int zynq_remoteproc_probe(struct platform_device *pdev)
 
 dma_mask_fault:
 	rproc_free(rproc);
+	if (cpu_up(1)) /* Cpu can't power on for example in nosmp mode */
+		dev_err(&pdev->dev, "Can't power on cpu1\n");
 
 	return ret;
 }
@@ -431,6 +434,7 @@ static int zynq_remoteproc_remove(struct platform_device *pdev)
 	struct rproc *rproc = platform_get_drvdata(pdev);
 	struct zynq_rproc_pdata *local = rproc->priv;
 	struct rproc_mem_entry *mem;
+  u32 ret;
 
 	dev_info(&pdev->dev, "%s\n", __func__);
 
@@ -446,6 +450,9 @@ static int zynq_remoteproc_remove(struct platform_device *pdev)
 	}
 
 	rproc_free(rproc);
+	ret = cpu_up(1);
+	if (ret)
+		dev_err(&pdev->dev, "Can't power on cpu1 %d\n", ret);
 
 	return 0;
 }
