diff -ur grpc_custom-v1.12.0/Makefile grpc_custom-v1.12.0.new/Makefile
--- grpc_custom-v1.12.0/Makefile        2018-05-15 17:10:37.000000000 +0000
+++ grpc_custom-v1.12.0.new/Makefile    2018-08-30 20:28:31.357410038 +0000
@@ -439,11 +439,11 @@
 DEFINES += $(EXTRA_DEFINES)
 LDLIBS += $(EXTRA_LDLIBS)

-HOST_CPPFLAGS = $(CPPFLAGS)
-HOST_CFLAGS = $(CFLAGS)
-HOST_CXXFLAGS = $(CXXFLAGS)
-HOST_LDFLAGS = $(LDFLAGS)
-HOST_LDLIBS = $(LDLIBS)
+HOST_CPPFLAGS += $(CPPFLAGS)
+HOST_CFLAGS += $(CFLAGS)
+HOST_CXXFLAGS += $(CXXFLAGS)
+HOST_LDFLAGS += $(LDFLAGS)
+HOST_LDLIBS += $(LDLIBS)

 # These are automatically computed variables.
 # There shouldn't be any need to change anything from now on.

